be/src/storage/binlog_manager.h

```diff
@@ -139,6 +140,18 @@ class BinlogManager : public std::enable_shared_from_this<BinlogManager> {
     // Commit the result of pre-commit, and it's visible for reading
     void commit_ingestion(int64_t version);
 
+    // Register the reader, and return a unique id allocated for this reader.
+    StatusOr<int64_t> register_reader(std::shared_ptr<BinlogReader> reader);
+
+    // Unregister the reader with the given id.
+    void unregister_reader(int64_t reader_id);
```

Comment:
what's the usage of register/unregister?

Reply:
notify these registered reader when removing expired files, and this logic will be in the next PR